src/group_theory/exponent.lean

```diff
@@ -55,6 +62,21 @@ def exponent_exists  := ∃ n, 0 < n ∧ ∀ g : G, g ^ n = 1
 noncomputable def exponent :=
 if h : exponent_exists G then nat.find h else 0
 
+variable {G}
+
+@[to_additive]
+lemma exponent_eq_zero_iff : exponent G = 0 ↔ ¬ exponent_exists G :=
+begin
+  rw [exponent],
+  split_ifs,
+  { simp [h, @not_lt_zero' ℕ] }, --if this isn't done this way, `to_additive` freaks
+  { tauto }
+end
+
+@[to_additive]
+lemma exponent_eq_zero_of_order_zero {g : G} (hg : order_of g = 0) : exponent G = 0 :=
```

Comment:
Should the name rather be `exponent_eq_zero_of_order_of_eq_zero`?

Reply:
I feel like in the rest of the file, `order` has been used interchangably with `order_of`; I can change it but I prefer snappier names (at least as much as possible within the naming conventions)

Reply:
Should we rename `order_of` to `order`? And update all the names? (Not in this PR, of course.)

Reply:
I'd be in favour of that!